src/core/Tree/Tree.tsx

```diff
@@ -0,0 +1,253 @@
+      />
+    )}
+  />
+ */
+
+export const Tree = <T,>(props: TreeProps<T>) => {
+  const { data, className, nodeRenderer, getNode, ...rest } = props;
+  useTheme();
+
+  const treeRef = React.useRef<HTMLUListElement>(null);
+
+  const focusedIndex = React.useRef<number>(0);
+  React.useEffect(() => {
+    focusedIndex.current = 0;
+  }, [data, getNode]);
+
+  const getFocusableNodes = React.useCallback(() => {
+    const focusableItems = getFocusableElements(treeRef.current);
+    // Filter out focusable elements that are inside each node, e.g. checkbox
+    return focusableItems.filter(
+      (i) => !focusableItems.some((p) => p.contains(i.parentElement)),
+    ) as HTMLElement[];
+  }, []);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLUListElement>) => {
+    const items = getFocusableNodes();
+    if (!items?.length) {
+      return;
+    }
+
+    const activeIndex = items.findIndex((el) =>
+      el.contains(treeRef.current?.ownerDocument.activeElement as HTMLElement),
+    );
+    const currentIndex = activeIndex > -1 ? activeIndex : 0;
+
+    switch (event.key) {
+      case 'ArrowUp': {
+        event.preventDefault();
+        const newIndex = Math.max(0, currentIndex - 1);
+        items[newIndex].focus();
+        focusedIndex.current = newIndex;
+        break;
+      }
+      case 'ArrowDown': {
+        event.preventDefault();
+        const newIndex = Math.min(items.length - 1, currentIndex + 1);
+        items[newIndex].focus();
+        focusedIndex.current = newIndex;
+        break;
+      }
+      default:
+        break;
+    }
+  };
+
+  const [flatNodesList, firstLevelNodesList] = React.useMemo(() => {
+    const flatList: FlatNode<T>[] = [];
+    const firstLevelNodes: FlatNode<T>[] = [];
+
+    const flattenNodes = (
+      nodes: T[] = [],
+      depth = 0,
+      parentNode?: FlatNode<T>,
+    ) => {
+      const nodeIdList = Array<string>();
+      nodes.forEach((element, index) => {
+        const { subNodes, ...nodeProps } = getNode(element);
+        const flatNode: FlatNode<T> = {
+          nodeProps,
+          depth,
+          parentNode,
+          indexInGroup: index,
+        };
+        nodeIdList.push(flatNode.nodeProps.nodeId);
+        flatList.push(flatNode);
+        if (depth === 0) {
+          firstLevelNodes.push(flatNode);
+        }
+        if (flatNode.nodeProps.isExpanded) {
+          const subNodeIds = flattenNodes(subNodes, depth + 1, flatNode);
+          flatNode.subNodeIds = subNodeIds;
+        }
+      });
+      return nodeIdList;
+    };
+
+    flattenNodes(data);
+
+    return [flatList, firstLevelNodes];
+  }, [data, getNode]);
+
+  return (
+    <ul
+      className={cx('iui-tree', className)}
+      role='tree'
+      onKeyDown={handleKeyDown}
+      ref={treeRef}
+      tabIndex={0}
+      style={{ outline: 0 }}
+      onFocus={() => {
+        const items = getFocusableNodes();
+        if (items.length > 0) {
+          items[focusedIndex.current]?.focus();
```

Comment:
This is also weird. How does it work? Because right now I'm seeing default browser outlines on the nodes.

Reply:
Dumb me, forgot that I need to add classes to `TreeNode`... Fixed.